### DISCUSSION TITLE
Remapのサイトでキーボードが認識しない

### DISCUSSION THREAD

**Discussion by palchan-net** [Yowkees/keyball#327] [Q&A]
Build Guide for Keyball39（https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp_conth.md）で「3-7．ProMicroの書き込みと動作確認」でサイトReMapからKeyballが認識されないため、それ以降の作業ができません。
おそらく半田付けが不十分なのだろうと思いますが、どのように切り分けしていくべきかわからず、ご教示いただければ幸いです。

まずProMicro、ダイオード、キーソケットのすべてが正常に接続できていて浮きがない状態でなければRemapで認識されないのでしょうか。
テスターを購入したので、ダイオード間の導通試験は行いましたが（おそらく接続できていると思いますが）、キーソケットはどのように導通試験すればいいのかわからずできておりません。

どのように調査すると良いのかご教示いただければ幸いです。
![image](https://github.com/Yowkees/keyball/assets/143974554/2f05a641-addd-4617-9f28-39752ccd95f3)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

書き込んでいただいたのは、「Keyball39_via」でしょうか。testファームではRemapで動作しませんので、ご注意ください。

ProMicroを基板から取り外して、ProMicro単体をパソコンにUSB接続するとどうでしょうか？
この場合リセットスイッチが押せませんので、下記リンクの画像のようにRSTとGNDをピンセット等でショートするとリセットできます。
https://kbd.dailycraft.jp/claw44/buildguide/firmware/

以上、お手数ですがご確認お願いいたします。

**Comment by palchan-net** [comment]
ProMicroが認識できず書き込みに失敗しているような状態です。まだテストファームの書き込みができておりません（最後にエラーとなっているので失敗している認識です、キースイッチのショートでも失敗します）
![image](https://github.com/Yowkees/keyball/assets/143974554/57edb49b-8707-46ba-9156-69cb81a97f4b)
ピンセットが通電することはテスターで確認の上、ピンでRSTとGNDをショートさせると、Keyballのリセットスイッチを押した時と同じ音（Windows上のUSBデバイスの認識音）がするので、リセットはできていると思います。
![image](https://github.com/Yowkees/keyball/assets/143974554/673a21b8-2237-4ad5-8395-e9a803f095f9)
本来はProMicro単体でファームウェアの書き込みが可能でしょうか。

**Comment by Yowkees** [comment]
ProMicro単体でもファームウェアの書き込みが可能です。
おそらくProMicroを２つご準備していただいていると思いますが、２つ共同じ症状でしょうか？

お手数ですがご確認お願いいたします。

**Reply by palchan-net** [reply]
> ProMicro単体でもファームウェアの書き込みが可能です。

という話を受けて、別のセットアップ直後のPCを用意して差し込んでみたらFirmwareの書き込みができました。（2個とも正常に認識→ファーム書き込み）
![image](https://github.com/Yowkees/keyball/assets/143974554/b26cab5a-9b5d-432b-aec5-4d6d60c16dc6)
![image](https://github.com/Yowkees/keyball/assets/143974554/d4397188-a4b2-4575-8dd2-bc389aeb00a7)

ピンセットでのキー入力動作確認も、一部だけですが動作確認できたので次の工程に進めそうです。

メインの作業端末で認識されない原因までは調べて居ませんが、状況としてはメインの作業端末だとProMicroが認識されず、別のPCに変えたら大丈夫でした。ご回答いただきありがとうございます。

**Comment by Yowkees** [comment]
PC側の問題でしたか！Alduinoのドライバー等見直していただければ幸いです。
以上、よろしくお願いいたします。

**Reply by palchan-net** [reply]
その後、リセットボタン2回での挙動もうまく扱えるようになり、作業端末側でもファームウェアの書き込みができるようになりました。（OSでのPnP認識一覧で差分を取りつつ）
また別の問題は発生している状況ではありますが、表題と異なる内容かつクローズ済みのようなので別で上げさせていただきます。
